src/Services/DeadlineDateService.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+class DeadlineDateService
+{
+    public static function checkDeadlineOverdue(?string $deadlineDate)
```

Comment:
This needs a unit test - however I feel like I remember some had written tests for it? Check with the team to see if anyone has already done this on another branch